source/common/upstream/upstream_impl.h

```diff
@@ -238,6 +238,13 @@ class HostDescriptionImplBase : virtual public HostDescription,
     last_hc_pass_time_.emplace(std::move(last_hc_pass_time));
   }
 
+  void setLbPolicyData(HostLbPolicyDataPtr lb_policy_data) override {
+    lb_policy_data_ = std::move(lb_policy_data);
+  }
+  OptRef<HostLbPolicyData> lbPolicyData() const override {
```

Comment:
This should probably not be a const method as it returns a reference to an internal object that may be changed later.
I do wonder if it will be possible to have the LbData in the `Host` class (without requiring more downcasts), but I don't really see that happening here.

We should also consider what will happen when multiple threads update the policy data concurrently.

Reply:
> We should also consider what will happen when multiple threads update the policy data concurrently.

The creation of LbData is on the main thread when the host is added. The content update of the LbData is on the worker threads to record some lb related stats from orca. But we require the custom LB to handle the thread safety by itself.
I will add some more clear comments.

Reply:
> I do wonder if it will be possible to have the LbData in the Host class (without requiring more downcasts), but I don't really see that happening here.

For now, it's impossible becase we can only access the host description from the upstream host.